python/pyarrow/_compute.pyx

```diff
@@ -1131,122 +1106,97 @@ cdef class _SplitPatternOptions(FunctionOptions):
-
+            c_sort_keys.push_back(CSortKey(tobytes(name),
+                                  unwrap_sort_order(order)))
         self.wrapped.reset(new CSortOptions(c_sort_keys))
 
 
 class SortOptions(_SortOptions):
-    def __init__(self, sort_keys=None):
-        if sort_keys is None:
-            sort_keys = []
+    def __init__(self, sort_keys):
         self._set_options(sort_keys)
 
 
 cdef class _SelectKOptions(FunctionOptions):
     def _set_options(self, k, sort_keys):
-        cdef:
-            c_string c_name
-            vector[CSortKey] c_sort_keys
-            CSortOrder c_order
-
+        cdef vector[CSortKey] c_sort_keys
         for name, order in sort_keys:
-            if order == "ascending":
-                c_order = CSortOrder_Ascending
-            elif order == "descending":
-                c_order = CSortOrder_Descending
-            else:
-                raise ValueError(
-                    "{!r} is not a valid order".format(order)
-                )
-            c_name = tobytes(name)
-            c_sort_keys.push_back(CSortKey(c_name, c_order))
-
+            c_sort_keys.push_back(CSortKey(tobytes(name),
+                                  unwrap_sort_order(order)))
         self.wrapped.reset(new CSelectKOptions(k, c_sort_keys))
 
 
 class SelectKOptions(_SelectKOptions):
-    def __init__(self, k, sort_keys=None):
-        if sort_keys is None:
-            sort_keys = []
+    def __init__(self, k, sort_keys):
         self._set_options(k, sort_keys)
 
 
 cdef class _QuantileOptions(FunctionOptions):
+    _interp_map = {
+        "linear": CQuantileInterp_LINEAR,
+        "lower": CQuantileInterp_LOWER,
+        "higher": CQuantileInterp_HIGHER,
+        "nearest": CQuantileInterp_NEAREST,
+        "midpoint": CQuantileInterp_MIDPOINT,
+    }
+
     def _set_options(self, quantiles, interp, skip_nulls, min_count):
-        interp_dict = {
-            'linear': CQuantileInterp_LINEAR,
-            'lower': CQuantileInterp_LOWER,
-            'higher': CQuantileInterp_HIGHER,
-            'nearest': CQuantileInterp_NEAREST,
-            'midpoint': CQuantileInterp_MIDPOINT,
-        }
-        if interp not in interp_dict:
-            raise ValueError(
-                '{!r} is not a valid interpolation'
-                .format(interp))
-        self.wrapped.reset(
-            new CQuantileOptions(quantiles, interp_dict[interp],
-                                 skip_nulls, min_count))
+        try:
+            self.wrapped.reset(
+                new CQuantileOptions(quantiles, self._interp_map[interp],
+                                     skip_nulls, min_count))
+        except KeyError:
+            _raise_invalid_function_option(interp, "quantile interpolation")
 
 
 class QuantileOptions(_QuantileOptions):
-    def __init__(self, *, q=0.5, interpolation='linear',
-                 skip_nulls=True, min_count=0):
+    def __init__(self, *, q=0.5, interpolation="linear", skip_nulls=True,
```

Comment:
```suggestion
    def __init__(self, q=0.5, *, interpolation="linear", skip_nulls=True,
```

? 

(since `q` is not a very descriptive keyword name anyway, it's not necessarily useful to have to spell it out)